onnxruntime/core/session/inference_session_utils.cc

```diff
@@ -0,0 +1,223 @@
+#include "core/session/inference_session_utils.h"
+
+namespace onnxruntime {
+
+//---------------------
+//--- local helpers ---
+//---------------------
+
+//--------------------------------------------
+//--- session options related helpers ---
+//--------------------------------------------
+// Below are some helpers that will be used to set corresponding session option values
+
+static Status
+SetIntraOpNumThreads(SessionOptions& session_options,
+                     int value,
+                     const logging::Logger& logger) {
+  if (value < 0) {
+    LOGS(logger, ERROR) << "Unsupported value for intra_op_num_threads: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for intra_op_num_threads: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting intra_op_num_threads to " << value;
+  session_options.intra_op_num_threads = value;
+  return Status::OK();
+}
+
+static Status SetInterOpNumThreads(SessionOptions& session_options,
+                                   int value,
+                                   const logging::Logger& logger) {
+  if (value < 0) {
+    LOGS(logger, ERROR) << "Unsupported value for inter_op_num_threads: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for inter_op_num_threads: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting inter_op_num_threads to " << value;
+  session_options.inter_op_num_threads = value;
+  return Status::OK();
+}
+
+static Status SetExecutionMode(SessionOptions& session_options,
+                               int value,
+                               const logging::Logger& logger) {
+  if (value != 0 && value != 1) {
+    LOGS(logger, ERROR) << "Unsupported execution_mode value in ORT config: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported execution_mode value in ORT config: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting execution_mode to " << (value == 0 ? "Sequential mode" : "Parallel mode");
+  session_options.execution_mode = (value == 0 ? ExecutionMode::ORT_SEQUENTIAL : ExecutionMode::ORT_PARALLEL);
+  return Status::OK();
+}
+
+static Status SetGraphOptimizationLevel(SessionOptions& session_options,
+                                        int value,
+                                        const logging::Logger& logger) {
+  switch (value) {
+    case 0:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_DISABLE_ALL";
+      session_options.graph_optimization_level = TransformerLevel::Default;
+      return Status::OK();
+
+    case 1:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_BASIC";
+      session_options.graph_optimization_level = TransformerLevel::Level1;
+      return Status::OK();
+
+    case 2:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_EXTENDED";
+      session_options.graph_optimization_level = TransformerLevel::Level2;
+      return Status::OK();
+
+    case 99:
```

Comment:
>99 [](start = 9, length = 2)

Shouldn't these values use the GraphOptimizationLevel values? i.e. instead of 99 use GraphOptimizationLevel::ORT_ENABLE_ALL

Reply:
Fixed.